aten/src/ATen/TensorIterator.cpp

```diff
@@ -345,12 +351,16 @@ void TensorIteratorBase::compute_types(const TensorIteratorConfig& config) {
 
   // Computes a common dtype, if needed
   if (has_different_input_dtypes && config.promote_inputs_to_common_dtype_) {
-    common_dtype_ = compute_common_dtype();
+    if (config.specified_common_dtype == ScalarType::Undefined) {
+      common_dtype_ = compute_common_dtype();
+    }
```

Comment:
else { } branch that checks if input tensors are promotable to the specified_common_dtype needed here